### DISCUSSION TITLE
Question: How for force deletion of a child QObject?

### DISCUSSION THREAD

**Discussion by rainman110** [MeVisLab/pythonqt#198] [Q&A]
Lets assume, I have a object hierarchy

Parent
 |--> child

`Parent` is currently managed by C++ and thus `child` as well. Now lets assume a python function `get_child`, that returns the child.

Now I'd like to make PythonQt the owner of the returned child object, meaning that when the returned object is deleted by python, also the child should be deleted and removed from its parent automatically. Here's the example to be more precise

```python
parent = get_parent()
child_obj = get_child(parent, own=True)

del child_obj # Now, child should be removed from parent
```

The function `PythonQtInstanceWrapper_deleteObject` is used as a destructor of QObject-type python objects. Inside this function, the code to free the underlying QObject is the following 

https://github.com/MeVisLab/pythonqt/blob/3cc771c6d79d7c03e758f87f3ca772d568d32af5/src/PythonQtInstanceWrapper.cpp#L105-L109

As you can see, the existence of a parent ensures, that the child is not deleted, even if PythonQt has ownership. This makes sense in most cases. This destructor function though has a `force` argument, which I'd like to use. How can I make sure, that the force argument is set to true?

A hint in the PythonQt code is

https://github.com/MeVisLab/pythonqt/blob/3cc771c6d79d7c03e758f87f3ca772d568d32af5/src/PythonQtInstanceWrapper.cpp#L344-L356

which forces object destruction, if `deleteSlot._type != PythonQtMemberInfo::Slot`. What does this mean? Is there any way to exploit this?

**Comment by usiems** [comment]
If I read this correctly this else branch should be used automatically if you didn't specify a "py_delete" slot method in your QObject.
(You can specify a py_delete method to handle deletion yourself.)

**Reply by rainman110** [reply]
Do you have any example how to use py_delete? So instead of calling the desctructor of the QObject, a custom py_delete slot is called to intercept the deletion?

But according to the code, without a "py_delete" slot, the branch `PythonQtInstanceWrapper_deleteObject(self, true); ` is called with force==true. I though did not observe, that deletion has been forced.
